inox2d/src/math/transform.rs

```diff
@@ -1,53 +1,8 @@
-use glam::{EulerRot, Mat4, Quat, Vec2, Vec3};
+use glam::{EulerRot, Mat4, Quat, Vec3};
 
-#[derive(Debug, Clone, Copy)]
-pub struct TransformOffset {
-	/// X Y Z
-	pub translation: Vec3,
-	/// Euler angles
-	pub rotation: Vec3,
-	/// X Y zoom
-	pub scale: Vec2,
-	/// Whether the transform should snap to pixels
-	pub pixel_snap: bool,
-}
-
-impl Default for TransformOffset {
-	fn default() -> Self {
-		Self {
-			translation: Vec3::ZERO,
-			rotation: Vec3::ZERO,
-			scale: Vec2::ONE,
-			pixel_snap: false,
-		}
-	}
-}
+use crate::node::data::TransformOffset;
```

Comment:
There is no reason to move `TransformOffset` to `crate::node::data`. It makes way more sense to expect the struct to be defined in `transform.rs`.
Just because a variant of node data uses it doesn't mean it makes more sense for it to be in that module.

Reply:
(See comment below). My direct response will be, just because a piece of node data require helper functions doesn't mean the data definition should be extracted outside and placed together with the helper functions.

E.g., `PhysicsState`

Reply:
What's crucial here is that it's fundamentally a piece of *math* just like the matrix, and just like all the `Vec2`s and `Vec3`s that we use and are provided by `glam`. The concept of a `Transform` is something really straightforward that isn't specific to Inochi2D, it's a generic concept in all sorts of graphics and we just happen to be using our own implementation. That is not the case of `Mask`, `Drawable`, `Part`, `Composite`, `PhysicsProps` and `SimplePhysics`, those are specific to Inochi2D's puppets.